File: frontend/src/api/aiHealth.ts
/**
 * API client functions for AI provider health dashboard.
 */

export interface ProviderCredits {
  remaining: number | null;
  total: number | null;
  unit: string;
}

export interface ProviderRateLimit {
  requests_per_minute: number | null;
  tokens_per_minute: number | null;
}

export interface ProviderHealthStatus {
  name: string;
  display_name: string;
  status: "reachable" | "unreachable" | "not_configured" | "error";
  is_default: boolean;
  error_message: string | null;
  credits: ProviderCredits | null;
  rate_limit: ProviderRateLimit | null;
  response_time_ms: number | null;
}

export interface AIHealthResponse {
  providers: ProviderHealthStatus[];
  default_provider: string;
}

const API_BASE = "/api/ai";

/**
 * Record a credits-exhausted / rate-limited response so the UI banner
 * (CreditsExhaustedBanner) can surface it. Called from any endpoint that
 * could trip OpenRouter's 402/429.
 */
function markCreditsIssue(status: number): void {
  if (status === 402 || status === 429) {
    sessionStorage.setItem("credits_exhausted", "true");
  }
}

/** Fetch health status for all AI providers. */
export async function fetchAIHealth(): Promise<AIHealthResponse> {
  const res = await fetch(`${API_BASE}/health`);
  if (!res.ok) {
    markCreditsIssue(res.status);
    throw new Error(`Failed to fetch AI health: ${res.status}`);
  }
  return res.json() as Promise<AIHealthResponse>;
}

/** Fetch health status for a single provider. */
export async function fetchProviderHealth(
  provider: string,
): Promise<ProviderHealthStatus> {
  const res = await fetch(`${API_BASE}/health/check?provider=${encodeURIComponent(provider)}`);
  if (!res.ok) {
    markCreditsIssue(res.status);
    throw new Error(`Failed to check provider health: ${res.status}`);
  }
  return res.json() as Promise<ProviderHealthStatus>;
}

File: src/career_os/schemas/ai_health.py
"""Pydantic schemas for AI provider health dashboard."""

from pydantic import BaseModel, Field

from career_os.schemas.constraints import INT64_MAX, INT64_MIN


class ProviderCredits(BaseModel):
    """Credit/quota information for an AI provider."""

    remaining: float | None = Field(None, description="Remaining credits/tokens")
    total: float | None = Field(None, description="Total credits/tokens quota")
    unit: str = Field("credits", description="Unit of measurement (credits, tokens, USD)")


class ProviderRateLimit(BaseModel):
    """Rate limit information for an AI provider."""

    requests_per_minute: int | None = Field(None, ge=INT64_MIN, le=INT64_MAX, description="RPM limit")
    tokens_per_minute: int | None = Field(None, ge=INT64_MIN, le=INT64_MAX, description="TPM limit")


class ProviderHealthStatus(BaseModel):
    """Health status for a single AI provider."""

    name: str = Field(..., description="Provider identifier")
    display_name: str = Field(..., description="Human-readable provider name")
    status: str = Field(
        ...,
        description="Health status: reachable | unreachable | not_configured | error",
    )
    is_default: bool = Field(False, description="Whether this is the currently active provider")
    error_message: str | None = Field(
        None, description="Error details if status is error/unreachable"
    )
    credits: ProviderCredits | None = Field(None, description="Credit/quota info if available")
    rate_limit: ProviderRateLimit | None = Field(None, description="Rate limit info if available")
    response_time_ms: float | None = Field(None, description="Ping response time in milliseconds")


class AIHealthResponse(BaseModel):
    """Response from GET /api/ai/health."""

    providers: list[ProviderHealthStatus] = Field(..., description="Health status per provider")
    default_provider: str = Field(..., description="Name of the current default provider")
